### PR TITLE
fix: update code and add tf latest version

### PR DESCRIPTION
**Pull request by mamrajyadav**
## what
- add terraform and azure provider latest version.
- fix tflint and tfsec issue.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,38 +1,50 @@
...
 
-##----------------------------------------------------------------------------- 
-## Below resource will deploy virtual network in your azure environment.    
+##-----------------------------------------------------------------------------
+## Below resource will deploy virtual network in your azure environment.
 ##-----------------------------------------------------------------------------
 resource "azurerm_virtual_network" "vnet" {
   count                   = var.enable == true ? 1 : 0
   name                    = format("%s-vnet", module.labels.id)
+  address_space           = length(var.address_spaces) == 0 ? [var.address_space] : var.address_spaces
```
Comment (13archit) [inline main.tf:27]: Can we use single variable in this, rather than using multiple variable?

File: main.tf [hunk 2]
```diff
@@ -1,38 +1,50 @@
...
   dns_servers             = var.dns_servers
   bgp_community           = var.bgp_community
   edge_zone               = var.edge_zone
-  flow_timeout_in_minutes = var.flow_timeout_in_minutes
+
+  encryption {
+    enforcement = var.enforcement
+  }
+
+  dynamic "subnet" {
```
Comment (13archit) [inline main.tf:39]: why this block is added?

File: _example/vnet_with_existing_ddos_id/example.tf [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+## Virtual Network module call.
+##-----------------------------------------------------------------------------
+module "vnet" {
+  source                 = "../../"
+  name                   = local.name
+  environment            = local.environment
+  resource_group_name    = module.resource_group.resource_group_name
+  location               = module.resource_group.resource_group_location
+  address_space          = "10.0.0.0/16"
+  existing_ddos_pp       = "/subscriptions/068245d4-3c94-42fe-9c4d-9e5e1cabc60c/resourceGroups/"
```
Comment (13archit) [inline _example/vnet_with_existing_ddos_id/example.tf:32]: Can you pls remove subscription id from here.

File: variables.tf [hunk 4]
```diff
@@ -124,49 +106,18 @@ variable "enable_network_watcher" {
...
   type        = string
-  default     = null
-  description = "Id of network security group for which flow are to be calculated"
-}
-
-variable "storage_account_id" {
-  type        = string
-  default     = null
-  description = "Id of storage account."
+  default     = "AllowUnencrypted"
```
Comment (13archit) [inline variables.tf:111]: Why it is set to unencrypted?

**Review by 13archit — COMMENTED** [review]
Fix Tfchecks.

File: main.tf (hunk 2 above)
Comment (d4kverma) [reply main.tf]: Specifies if the encrypted Virtual Network allows VM that does not support encryption

File: variables.tf (hunk 4 above)
Comment (d4kverma) [reply variables.tf:105]: "This subscription does not support \"DropUnencrypted\" as enforcement policy on encrypted vnet. Please register \"Microsoft.Network/AllowDropUnecryptedVnet\" feature on this subscription." Details=[]
│

**Review by d4kverma — APPROVED** [review]
LGTM

**Pull request merged**